src/pages/cw-storage-plus/containers/indexed-map.mdx

```diff
@@ -0,0 +1,244 @@
+    to create the index.
+  - A prefix. The index needs its own storage namespace. Note the index prefix
+    has to be distinct from the `IndexedMap` prefix (or any other prefix used
+    with this contract's storage).
+- We construct an `IndexedMap` with that list of indexes.
+
+<Callout type="warning">
+If you're using
+  [`UniqueIndex`](https://docs.rs/cw-storage-plus/latest/cw_storage_plus/struct.UniqueIndex.html),
+  it's your responsibility to ensure that the index you've built has unique
+  keys. If you have two users with the same handle, the index will be
+  overwritten and only contain the last user. Be careful!
+  
+  If you need to store a key that is not unique, you'll want to use a [`MultiIndex`](https://docs.rs/cw-storage-plus/latest/cw_storage_plus/struct.MultiIndex.html) - see the [next section](#lookup-by-country).
+</Callout>
+
+<Callout>
+Under the hood, the `IndexedMap` will store the data using a regular `Map` for
+the primary key, and then another `Map` for each secondary index. This is how efficient lookups are achieved.
+
+Again, every update to this storage structure will have to update all the
+indexes - that's the price we pay for efficient lookups.
+
+</Callout>
+
+### Taking advantage of the secondary index
+
+As with a regular `Map`, you can
+[load data](https://docs.rs/cw-storage-plus/latest/cw_storage_plus/struct.IndexedMap.html#method.load),
+[save](https://docs.rs/cw-storage-plus/latest/cw_storage_plus/struct.IndexedMap.html#method.save)
+data,
+[remove](https://docs.rs/cw-storage-plus/latest/cw_storage_plus/struct.IndexedMap.html#method.remove)
+data, and iterate over the `IndexedMap`.
+
+```rust template="storage"
+use cw_storage_plus::IndexedMap;
+
+use crate::users::*;
+
+let users = IndexedMap::<Addr, _, _>::new("u", user_indexes("uh"));
+
+let alice_addr = Addr::unchecked("alice");
+let alice_user = User {
+    handle: "alice".to_string(),
+    country: "Wonderland".to_string(),
+};
+
+let bob_addr = Addr::unchecked("bob");
+let bob_user = User {
+    handle: "bob".to_string(),
+    country: "Bikini Bottom".to_string(),
+};
+
+users
+    .save(&mut storage, alice_addr.clone(), &alice_user)
+    .unwrap();
+
+users
+    .save(&mut storage, bob_addr.clone(), &bob_user)
+    .unwrap();
+
+assert_eq!(
+    users.load(&storage, Addr::unchecked("alice")).unwrap(),
+    alice_user
+);
+
+assert_eq!(
+    users
+        .range(&storage, None, None, Order::Ascending)
+        .collect::<StdResult<Vec<_>>>()
+        .unwrap(),
+    vec![
+        (alice_addr.clone(), alice_user.clone()),
+        (bob_addr.clone(), bob_user.clone()),
+    ]
+);
+```
+
+But now we can also look up users by their handle.
+
+```rust template="storage"
+use cw_storage_plus::IndexedMap;
+
+use crate::users::*;
+
+let (_key, alice) = users
+    .idx
+    .handle_ix                           // access the `handle_ix` index
+    .item(&storage, "alice".to_string()) // load a specific record
```

Comment:
I was a _bit_ confused by this - saw "alice" and thought maybe we're looking up the address... totally my mistake, but I'd suggest for the sake of example to give alice and bob different strings for the address vs. handle

Reply:
Yeah, good call